File: Interviews/GFG/Daily-Problems/2022/October/Day-16/Move-Last-Element-to-Front-of-a-Linked-List.cpp
class Solution {
public:
    ListNode *moveToFront(ListNode *head) {
        ListNode *cur = head, *pre = NULL;
        while (cur->next) {
            pre = cur;
            cur = cur->next;
        }

        if (!pre) return head;
        pre->next = NULL;
        cur->next = head;
        return cur;
    }
};

File: Interviews/GFG/random-problems/Easy/Move-Last-Element-to-Front-of-a-Linked-List/Move-Last-Element-to-Front-of-a-Linked-List.py
class Solution:
    def moveToFront(self, head : Optional['Node']) -> Optional['Node']:
        cur, pre = head, None
        while cur.next:
            pre = cur
            cur = cur.next
        
        if not pre:
            return head
        pre.next = None
        cur.next = head
        return cur